app/Http/Controllers/Users/BulkUsersController.php

```diff
@@ -76,6 +77,33 @@ public function edit(Request $request)
                 }
                 return redirect()->back()->with('success', trans('admin/users/message.password_resets_sent'));
 
+            } elseif ($request->input('bulk_actions') == 'print') {
+                $users = User::query()
+                    ->with([
+                        'assets.assetlog',
+                        'assets.assignedAssets.assetlog',
+                        'assets.assignedAssets.defaultLoc',
+                        'assets.assignedAssets.location',
+                        'assets.assignedAssets.model.category',
+                        'assets.defaultLoc',
+                        'assets.location',
+                        'assets.model.category',
+                        'accessories.assetlog',
+                        'accessories.category',
+                        'accessories.manufacturer',
+                        'consumables.assetlog',
+                        'consumables.category',
+                        'consumables.manufacturer',
+                        'licenses.category',
+                    ])
+                    ->withTrashed()
+                    ->findMany($request->input('ids'));
+
+                $users->each(fn($user) => $this->authorize('view', $user));
```

Comment:
Not positive about this since the top of the controller method has `$this->authorize('update', User::class);`...might end up in a place where the user has access to view users but not update them?

Or maybe I should make this `$this->authorize('view', User::class);`?

Reply:
I think we can start with `$this->authorize('view', User::class);` and add the stronger gates within that faux-router controller method.